libsplinter/src/service/scabbard/client/mod.rs

```diff
@@ -0,0 +1,306 @@
+        self,
+        org_id: &str,
+        name: &str,
+        smart_permission_file: &str,
+    ) -> Result<Self, Error> {
+        self.create_smart_permission(org_id, name, load_file_into_bytes(smart_permission_file)?)
+    }
+
+    /// Queue a transaction to create a smart permission, with the specified
+    /// `smart_permission_function`.
+    pub fn create_smart_permission(
+        mut self,
+        org_id: &str,
+        name: &str,
+        smart_permission_function: Vec<u8>,
+    ) -> Result<Self, Error> {
+        self.unsent_payloads
+            .push(smart_permission::create_smart_permission_creation_payload(
+                org_id,
+                name,
+                smart_permission_function,
+            )?);
+        Ok(self)
+    }
+
+    /// Queue a transaction to update a smart permission, loaded from the specified
+    /// `smart_permission_file`.
+    pub fn update_smart_permission_from_file(
+        self,
+        org_id: &str,
+        name: &str,
+        smart_permission_file: &str,
+    ) -> Result<Self, Error> {
+        self.update_smart_permission(org_id, name, load_file_into_bytes(smart_permission_file)?)
+    }
+
+    /// Queue a transaction to update a smart permission, with the specified
+    /// `smart_permission_function`.
+    pub fn update_smart_permission(
+        mut self,
+        org_id: &str,
+        name: &str,
+        smart_permission_function: Vec<u8>,
+    ) -> Result<Self, Error> {
+        self.unsent_payloads
+            .push(smart_permission::create_smart_permission_update_payload(
+                org_id,
+                name,
+                smart_permission_function,
+            )?);
+        Ok(self)
+    }
+
+    /// Queue a transaction to delete a smart permission.
+    pub fn delete_smart_permission(mut self, org_id: &str, name: &str) -> Result<Self, Error> {
+        self.unsent_payloads
+            .push(smart_permission::create_smart_permission_delete_payload(
+                org_id, name,
+            )?);
+        Ok(self)
+    }
+
+    /// Load the signing key in the given key file.
+    pub fn load_signing_key_from_file(filename: &str) -> Result<Secp256k1PrivateKey, Error> {
+        key::load_signing_key_from_file(filename)
+    }
+}
+
+/// Load the contents of a file into a bytes vector.
+fn load_file_into_bytes(payload_file: &str) -> Result<Vec<u8>, Error> {
+    let file = File::open(payload_file)
+        .map_err(|err| Error::new_with_source("failed to load file", err.into()))?;
+    let mut buf_reader = BufReader::new(file);
+    let mut contents = Vec::new();
+    buf_reader
+        .read_to_end(&mut contents)
+        .map_err(|err| Error::new_with_source("failed to read file", err.into()))?;
+    Ok(contents)
+}
+
+#[cfg(test)]
+mod tests {
+    use tempdir::TempDir;
+
+    pub const MOCK_PRIV_KEY_HEX: &str =
+        "d31e395bed0d9b2277b25d57523063d7d6b9db802d80549bc1362875cdcb83c6";
+
+    pub fn new_temp_dir() -> TempDir {
```

Comment:
Not sure I see a huge value in not just copying this code to where it used, especially since it doesn't look like you are always sharing the MOCK key.

Reply:
Updated